nixos/modules/services/hardware/nvidia-container-toolkit/default.nix

```diff
@@ -289,11 +293,52 @@
 
       systemd.services.nvidia-container-toolkit-cdi-generator = {
         description = "Container Device Interface (CDI) for Nvidia generator";
+        requiredBy = lib.mkMerge [
+          (lib.mkIf config.virtualisation.docker.enable [ "docker.service" ])
+          (lib.mkIf config.virtualisation.podman.enable [ "podman.service" ])
+        ];
         wantedBy = [ "multi-user.target" ];
-        after = [ "systemd-udev-settle.service" ];
         serviceConfig = {
           RuntimeDirectory = "cdi";
           RemainAfterExit = true;
+          ExecStartPre = pkgs.writeShellScript "wait-for-nvidia-devices" ''
```

Comment:
Isn't this something like `ConditionPathExistsGlob=/dev/nvidia*`? Or, probably, more like a `systemd.paths."...".pathConfig.PathExistsGlob` I guess so it actually watches for changes?

Reply:
That's an interesting point thanks, I'll check and report back